Assets/HoloToolkit/Input/Scripts/Cursor/InteractiveMeshCursor.cs

```diff
@@ -153,15 +155,38 @@ public override void SetVisiblity(bool visible)
 
             if (visible)
             {
+                if (Light != null)
```

Comment:
Good points. I was over thinking it because it feels counter productive to set the visibility one way if I know for sure the visibility will be set back again in the same frame. I removed all the extra checks so the sequence now works like this.
1. SetVisibility true/false
2. The elements' visibility is set
3. If visibility is true, then set the state change which will set the visibility of the ring or dot depending on state.